components/schedule-editor/src/ScheduleEditor.svelte

```diff
@@ -248,10 +288,165 @@
   }
 
   function addNewField(field: CustomField) {
-    _this.custom_fields = [..._this.custom_fields, field];
+    const fieldWithId = {
+      ...field,
+      id: String(_this.custom_fields.length),
+    };
+    _this.custom_fields = [..._this.custom_fields, fieldWithId];
+
     newCustomField = { ...emptyCustomField };
+
     newFieldTitleElement.focus();
   }
+
+  type CustomFieldWithId = CustomField & { id: string };
+  let draggedField: CustomFieldWithId | null = null;
+  let draggedFieldIndex: number;
+  let tablerowRect: DOMRect;
+  let tablecellRect: DOMRect[] = []; // To replicate cell dimensions on drag-placeholder
+  type CustomFieldRefs = Record<string, HTMLElement>;
+  let customFieldRefs: CustomFieldRefs = {}; // store the refs to dom nodes once so we can recalculate sizes on the fly
+  let customFieldDomRects: DOMRect[] = [];
+  let customFieldIdOrder: string[] = [];
+  let maxDragTop: number;
+  let maxDragBottom: number;
+  let dragBegins: boolean = false;
+  let dragPlaceholder = {
+    left: 0,
+    top: 0,
+  };
+
+  type CustomFieldDrag = {
+    event: MouseEvent;
+    field: CustomFieldWithId;
+    i: number;
+  };
+
+  function handleCustomFieldDrag({ event, field, i }: CustomFieldDrag) {
+    if (!dragBegins) {
+      customFieldDomRects = getDomRects(customFieldRefs);
+      dragBegins = true;
+    }
+
+    draggedFieldIndex = i;
+    draggedField = field;
+
+    setDragRowPosition(event);
+  }
+
+  function handleCustomFieldDragRelease() {
+    if (draggedField) {
+      draggedField = null;
+    }
+  }
+
+  function swapHoveredWithDraggedFields(e: MouseEvent) {
+    const hoveredFieldIndex = customFieldDomRects.findIndex(
```

Comment:
Do you think it would be easier to track the current hovered field during a drag using mouse enter/leave events on the other items?

Reply:
Kinda touched on this on [a reply to Phil](https://github.com/nylas/components/pull/220#issuecomment-981747191), but will take another look into this approach. Thanks for bringing it up!